src/main/java/net/sf/mzmine/datamodel/impl/SimpleDataPoint.java

```diff
@@ -59,6 +59,10 @@ public double getMZ() {
 	return mz;
     }
 
+    public void setIntensity(final double intensity) {
```

Comment:
The setIntensity() method should not be here. Data points are immutable by design (as described in the class javadoc), same as other basic Java classes (String, Double, etc.). Instead of changing the intensity, you should always create a new data point instance.